src/components/SignupForm/index.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+
+import logo from '../../assets/logo.png';
+import ButtonForm from '../ButtonForm';
+
+import './SignupForm.scss';
+
+function SignupForm() {
+  const { register, handleSubmit, errors, getValues } = useForm();
+
+  const isPasswordEqual = (password: string) => password === getValues('password');
```

Comment:
I suggest you move the validators to a utilities file, so that in future forms you can reuse them and make use of them again